brain_friendly.py

```diff
@@ -45,6 +60,29 @@ def decrement_command(value):
     else:
         return value - 1
 
+def get_brace_matches(program):
+    '''
+    Returns a dictionary containing the pairs for left and right brace indexes.
+    If the program is malformed, then ValueError is raised.
+    '''
+    brace_dict = dict()
+    brace_stack = list()
+
+    for index in range(len(program)):
+        if program[index] == "[":
+            brace_stack.append(index)
+        elif program[index] == "]":
+            if len(brace_stack) == 0: #sanity check
+                raise ValueError("Malformed BF program (too many \"]\"s)")
+            left_index = brace_stack.pop()
+            #add left AND right brace pairs.
+            brace_dict[left_index] = index
+            brace_dict[index] = left_index
+
+    if len(brace_stack) == 0: #sanity check
```

Comment:
For style, this check should be `if not brace_stack:`, but this would actually probably be better as
```python
if brace_stack:
    raise ValueError('Malformed BF program (unmatched "["')
return brace_dict
```